examples/ota-provider-app/linux/README.md

```diff
@@ -10,13 +10,81 @@ Suggest doing the following:
 
 ## Usage
 
-`./ota-provider-app [-f/--filepath \<file\>]`
+`./ota-provider-app [-f/--filepath \<file\>] [-o/--otaImageList \<file\>]`
 
 If `--filepath` is supplied, `ota-provider-app` will automatically serve that
 file to the OTA Requestor (SoftwareVersion will be Requester version + 1).
 
-If no `--filepath` is supplied, `ota-provider-app` will respond to `QueryImage`
-with `NotAvailable` status.
+If `--otaImageList` is supplied, `ota-provider-app` will parse the JSON file and
+extract all required data. Then the most recent, valid software version will be
+selected and the corresponding ota-file will be sent to the OTA Requestor.
+
+
+Here's an example of the otaImageList file contents:
+
+{
+    "foo": 1, // ignored by parser
+    "deviceSoftwareVersionModel": [
+        {
+            "vendorId": 1,
+            "productId": 1,
+            "softwareVersion": 10,
+            "softwareVersionString": "1.0.0",
+            "CDVersionNumber": 18,
+            "softwareVersionValid": true,
+            "minApplicableSoftwareVersion": 0,
+            "maxApplicableSoftwareVersion": 100,
+            "otaURL": "/tmp/ota.txt"
+        },
+        {
+            "vendorId": 1,
+            "productId": 1,
+            "softwareVersion": 20,
+            "softwareVersionString": "1.0.1",
+            "CDVersionNumber": 18,
+            "softwareVersionValid": false,
+            "minApplicableSoftwareVersion": 0,
+            "maxApplicableSoftwareVersion": 100,
+            "otaURL": "/tmp/ota.txt"
+        },
+        {
+            "vendorId": 1,
+            "productId": 1,
+            "softwareVersion": 30,
+            "softwareVersionString": "1.0.2",
+            "CDVersionNumber": 18,
+            "softwareVersionValid": true,
+            "minApplicableSoftwareVersion": 0,
+            "maxApplicableSoftwareVersion": 100,
+            "otaURL": "/tmp/ota.txt"
+        },
+        {
+            "vendorId": 1,
+            "productId": 1,
+            "softwareVersion": 40,
+            "softwareVersionString": "1.1.0",
+            "CDVersionNumber": 18,
+            "softwareVersionValid": true,
+            "minApplicableSoftwareVersion": 0,
+            "maxApplicableSoftwareVersion": 100,
+            "otaURL": "/tmp/ota.txt"
+        },
+        {
+            "vendorId": 1,
+            "productId": 1,
+            "softwareVersion": 50,
+            "softwareVersionString": "1.1.1",
+            "CDVersionNumber": 18,
+            "softwareVersionValid": false,
+            "minApplicableSoftwareVersion": 0,
+            "maxApplicableSoftwareVersion": 100,
+            "otaURL": "/tmp/ota.txt"
+        }
+    ]
+}
+
+If neither `--filepath` nor `--otaImageList` are supplied, `ota-provider-app`
```

Comment:
Probably worth mentioning that both cannot/should not be supplied

Reply:
I added logic in the argparser to throw a runtime error if both args are supplied, with an appropriate message.